src/js/base/routing.md

```
@@ -0,0 +1,173 @@
+  definition: { action, route, root, meta },
+}
+```
+
+- `getCurrentRoute()` returns this object; `getCurrentRouteMeta()` returns its `meta`.
+- `before:appRoute` and `appRoute` receive the context object (not positional args).
+- Action handlers still receive **positional** arguments (`showPatient(patientId)`).
+
+Side-effect order inside `routeAction` (deliberate — the current route is set first
+so it is observable in `before:appRoute`):
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)